### ISSUE TITLE
Android: Adding custom actions

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30854]
## Description
The `accessibilityActions` and `onAccessibilityAction` provided by React Native (https://reactnative.dev/docs/accessibility#accessibility-actions) do not work on all core components.  

Specifically Button, TextInput, Text, and Picker do not allow these properties to be set, even though they do support other accessibility properties.

Custom actions are often added to help streamline the UI, and allowing them to be used in some places but not others limits their usefulness.

## React Native version:
v0.63

## Expected Behavior
AccessibilityActions and the onAccessibilityAction callback should be able to work on all core components.

## Android Details
AccessibilityActions and onAccessibilityAction are already well implemented on the Android side, so this change likely only needs to happen in the components themselves on the Javascript side.

**Comment by dennisurtubia** [comment]
Hi, @amarlette! Can I work on this issue? Thank you for now!

**Comment by amarlette** [comment]
Thank you for taking this on @dennisurtubia! I'll move this into the "In-Progress" column

**Comment by amarlette** [comment]
Closed by #31532
